src/protocols/common/CommonProtocol.h

```diff
@@ -0,0 +1,82 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      The defines constants for the CHIP Common Protocol, present in
+ *      every CHIP device.
+ *
+ */
+
+#ifndef _COMMON_PROTOCOL_H
+#define _COMMON_PROTOCOL_H
+
+/**
+ *   @namespace chip::Protocols::Common
+ *
+ *   @brief
+ *     This namespace includes all interfaces within CHIP for the
+ *     CHIP Common protocol.
+ *
+ *     The interfaces define message types and status codes.
+ */
+
+namespace chip {
+namespace Protocols {
+namespace Common {
+
+/**
+ * Common Protocol Message Types
+ */
+enum
+{
+    kMsgType_StatusReport = 1,
+    kMsgType_Null         = 2,
+};
+
+/**
+ * Common Protocol Status Codes
+ */
+enum
+{
+    kStatus_Success                = 0,      /**< The operation completed without error. */
+    kStatus_Canceled               = 1,      /**< The operation was canceled. */
+    kStatus_BadRequest             = 0x0010, /**< The request was unrecognized or malformed. */
+    kStatus_UnsupportedMessage     = 0x0011, /**< An unrecognized or unsupported message was received. */
+    kStatus_UnexpectedMessage      = 0x0012, /**< A message was received at an unexpected time or in an unexpected context. */
+    kStatus_AuthenticationRequired = 0x0013, /**< The request can only be made by an authenticated party. */
+    kStatus_AccessDenied           = 0x0014, /**< The request can only be made by a party with sufficient access. */
+    kStatus_OutOfMemory =
+        0x0017, /**< The sender is low on memory resources and cannot perform the requested operation at the current time. */
+    kStatus_NotAvailable       = 0x0018, /**< The requested operation cannot be performed given the current state of the sender. */
+    kStatus_LocalSetupRequired = 0x0019, /**< The requested operation could not be performed because one or more necessary local
+                                            configuration steps have not been completed. */
+    kStatus_InternalServerProblem = 0x0020, /**< Request could not be completeted because of problems on the server. */
+    kStatus_Relocated             = 0x0030, /**< Request was made to the wrong endpoint. Client should query its
+                                                 directory server for an updated endpoint list and try again. */
+    kStatus_Busy          = 0x0040, /**< The sender is busy and cannot perform the requested operation at the current time. */
+    kStatus_Timeout       = 0x0041, /**< The operation or protocol interaction failed to complete in the allotted time. */
+    kStatus_InternalError = 0x0050, /**< An internal failure prevented an operation from completing. */
+    kStatus_Continue      = 0x0090, /**< Context-specific signal to proceed. */
+};
+
+} // namespace Common
```

Comment:
is the "Common" namespace needed?

Reply:
each protocol is under its own namespace, even Common protocol defines the messages shared by all protocols, but its has it own dedicated protocol ID, so I prefer keep "Common" namespace.

Reply:
Does this need to be harmonized with the work @mlepage-google is working on. I've seen some namespace spec work referring to a `std` namespace which is likely synonymous with `Common`. Either way, I think there very much is utility for having a `Project CHIP Common / Standard` namespace / vendor.

Reply:
In general, we've been using 0x0000 (or similar zero) for special namespace values for common/CHIP/standard (it won't be CHIP because it'll have a different name by then). The namespace draft spec mostly deals with cluster identifiers and device type identifiers, but that concept may apply here as well. In TLV the leading zeroes will melt away.

Reply:
> In general, we've been using 0x0000 (or similar zero) for special namespace values for common/CHIP/standard (it won't be CHIP because it'll have a different name by then). The namespace draft spec mostly deals with cluster identifiers and device type identifiers, but that concept may apply here as well. In TLV the leading zeroes will melt away.

I agree, 0x0000 should not be used for kChipProtocol_SecurityChannel, but this is defined and approved by the spec

https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/secure_channel/Secure_Channel_Protocol.adoc

Secure Channel Protocol Id = 0x0000.

I think we need to update the spec first,

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/288 to track this change.

Reply:
Synced with Martin, Secure Channel Protocol Id  will be updated to 0x0001 in spec, and I also removed Common protocol since this is not generally agreed in SCTT.  The current proposal is no generic messages used across protocols and status would be in Data Model.